### DISCUSSION TITLE
Scatterplot with dasharray stroke?

### DISCUSSION THREAD

**Discussion by chrisgervang** [visgl/deck.gl#4512] [Q&A]
Can I create a scatterplot element with an dasharray outline, similar to the feature compatible with PathLayer? Here's a circle of the idea in Google Slides.

![Screen Shot 2020-04-17 at 4 18 38 PM](https://user-images.githubusercontent.com/2461547/79621219-6afc7a80-80c7-11ea-84ff-041a88e3c075.png)

I found a [`PathStyleExtension`](https://deck.gl/#/documentation/submodule-api-reference/deckgl-extensions/path-style-extension?section=installation) that provides a getDashArray prop, but I'm unsure how to apply that to a scatter plot stroke since it isn't a PathLayer.

**Comment by Pessimistress** [comment]
You can subclass the `ScatterplotLayer` with a custom fragment shader.

Assume you send a custom uniform `dashArray`:

```glsl
uniform vec2 dashArray;
```

After this line:

https://github.com/visgl/deck.gl/blob/82dbe806712b2a10da11d86a5c48e2dadce1eb68/modules/layers/src/scatterplot-layer/scatterplot-layer-fragment.glsl.js#L46

Add something along the lines of:

```glsl
float circumferencePixels = outerRadiusPixels * PI * 2.0;
float numOfDashes = floor(circumferencePixels / (dashArray.x + dashArray.y));
float anglePerDash = PI * 2.0 / numOfDashes;
float angle = atan(geometry.uv.y, geometry.uv.x);
bool isInGap = fract(angle/anglePerUnit) > dashArray.x / (dashArray.x + dashArray.y);
if (isInGap) discard;
```
